index.html

```diff
@@ -3549,28 +3548,45 @@ <h3><dfn>Writing or pushing content</dfn></h3>
         the <a>activated reader objects</a>, run the following sub-steps:
         <ol>
           <li>
-            If |reader|.<a>[[\Id]]</a> is [= dictionary member/present =]
-            and it is not equal to any |record|'s
-            <a href="#dom-ndefrecord-id">id</a> where |record|
-            is an element of |message|, [= iteration/continue =].
-          </li>
-          <li>
-            If |reader|.<a>[[\RecordType]]</a> is [= dictionary member/present =]
-            and it is not equal to any |record|'s
-            <a href="#dom-ndefrecord-recordtype">recordType</a> where |record|
-            is an element of |message|, [= iteration/continue =].
-          </li>
-          <li>
-            If |reader|.<a>[[\MediaType]]</a> is not `""` and
-            it is not equal to any |record|'s
-            <a href="#dom-ndefrecord-mediatype">mediaType</a> where |record| is
-            an element of |message|, [= iteration/continue =].
-          </li>
-          <li>
-            <a>Fire an event</a> named "`reading`" at |reader|
-            using <a>NDEFReadingEvent</a> with its <a>serialNumber</a> attribute
-            initialized to |serialNumber| and <a>message</a> attribute
-            initialized to |message|.
+            [= list/For each =] |record:NDEFRecord| in |message:NDEFMessage|,
+            <ol>
+              <li>
+                Let |matched:boolean| be `false`.
+              </li>
+              <li>
+                If |reader|.<a>[[\Id]]</a> is not `""` and is not equal to
+                |record|'s <a href="#dom-ndefrecord-id">id</a>,
+                [= iteration/continue =].
+                If it is equal, set |matched| to `true`.
```

Comment:
Why day if equal, you continue above if not

Reply:
Maybe

If "" continue
If not equal continue
Set matched to true

Reply:
If empty, continue is not good because you want to check for the next fields.

Reply:
I have played a bit with other formulations (in the spirit of your comment above) but I find this more clear. If a filter slot is not specified, match depends on other slots. If defined, and not matched, we can continue (otherwise depends on other slots again, but so far is a match).

Reply:
How about this pattern below?

> - If reader.id is not empty string and is equal to records' id, set matched to true.
> - If matched is false, continue.

Reply:
That doesn't work since if reader.id was empty, it's will depend on the others, so we don't continue, but fall through.

Reply:
Let's move on. We may want to ask @leonhsl POV when it gets implemented.

Reply:
FYI, [the current implementation](https://source.chromium.org/chromium/chromium/src/+/master:services/device/nfc/android/java/src/org/chromium/device/nfc/NfcImpl.java;l=597) is based on the consensus we got on the spec issue #437.